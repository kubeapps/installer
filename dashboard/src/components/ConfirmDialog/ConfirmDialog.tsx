import { CdsButton } from "@cds/react/button";
import { CdsModal, CdsModalActions, CdsModalContent, CdsModalHeader } from "@cds/react/modal";
import Alert from "components/js/Alert";
import React from "react";
import LoadingWrapper from "../LoadingWrapper/LoadingWrapper";
import "./ConfirmDialog.css";

interface IConfirmDialogProps {
  modalIsOpen: boolean;
  loading: boolean;
  extraElem?: JSX.Element;
  headerText?: string;
  confirmationText: string;
  confirmationButtonText?: string;
  error?: Error;
  onConfirm: () => any;
  closeModal: () => any;
}

function ConfirmDialog({
  modalIsOpen,
  loading,
  extraElem,
  headerText,
  confirmationButtonText,
  confirmationText,
  onConfirm,
  closeModal,
  error,
}: IConfirmDialogProps) {
  return (
    <>
      {modalIsOpen && (
        <CdsModal closable={true} onCloseChange={closeModal}>
          <CdsModalHeader>{headerText}</CdsModalHeader>
          {error && <Alert theme="danger">An error ocurred: {error.message}</Alert>}
          {loading === true ? (
            <>
              <CdsModalContent>
                <span>Loading, please wait</span>
                <LoadingWrapper loaded={false} />
              </CdsModalContent>
            </>
          ) : (
            <>
              <CdsModalContent>
                <p>{confirmationText}</p>
                <p>{extraElem}</p>
              </CdsModalContent>
              <CdsModalActions>
                <CdsButton action="outline" type="button" onClick={closeModal}>
                  Cancel
                </CdsButton>
                <CdsButton status="danger" type="submit" onClick={onConfirm}>
                  {confirmationButtonText || "Delete"}
                </CdsButton>
              </CdsModalActions>
            </>
          )}
        </CdsModal>
      )}
    </>
  );
}

export default ConfirmDialog;
