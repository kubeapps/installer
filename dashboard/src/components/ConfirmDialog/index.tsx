import * as React from "react";
import * as Modal from "react-modal";

interface IConfirmDialogProps {
  modalIsOpen: boolean;
  loading: boolean;
  onConfirm: () => Promise<any>;
  closeModal: () => Promise<any>;
}

interface IConfirmDialogState {
  modalIsOpen: boolean;
}

class ConfirmDialog extends React.Component<IConfirmDialogProps, IConfirmDialogState> {
  public state: IConfirmDialogState = {
    modalIsOpen: this.props.modalIsOpen,
  };

  public render() {
    return (
      <div className="ConfirmDialog">
        <Modal
          style={{
            content: {
              bottom: "auto",
              left: "50%",
              marginRight: "-50%",
              right: "auto",
              top: "50%",
              transform: "translate(-50%, -50%)",
            },
          }}
          ariaHideApp={false}
          isOpen={this.props.modalIsOpen}
          onRequestClose={this.closeModal}
          contentLabel="Modal"
        >
          {this.props.loading === true ? (
            <div> Loading ... </div>
          ) : (
            <div>
              <div> Are you sure you want to delete this? </div>
              <button id="cancel" className="button" onClick={this.props.closeModal}>
                Cancel
              </button>
              <button
                id="delete"
                className="button button-primary button-danger"
                type="submit"
                onClick={this.props.onConfirm}
              >
                Delete
              </button>
            </div>
          )}
        </Modal>
      </div>
    );
  }

  public closeModal = () => {
    this.setState({
      modalIsOpen: false,
    });
  };
}

export default ConfirmDialog;
