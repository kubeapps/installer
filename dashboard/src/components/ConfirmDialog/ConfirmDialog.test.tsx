import { CdsModalContent } from "@cds/react/modal";
import { mount } from "enzyme";
import context from "jest-plugin-context";
import itBehavesLike from "../../shared/specs";
import ConfirmDialog from "./ConfirmDialog";

const defaultProps = {
  loading: false,
  modalIsOpen: true,
  onConfirm: jest.fn(),
  closeModal: jest.fn(),
};

context("when loading is true", () => {
  itBehavesLike("aLoadingComponent", {
    component: ConfirmDialog,
    props: { ...defaultProps, loading: true },
  });
});

it("should modify the default confirmation text", () => {
  const wrapper = mount(
    <ConfirmDialog {...defaultProps} confirmationText="Sure?" confirmationButtonText="Sure!" />,
  );
  expect(wrapper.find(CdsModalContent)).toIncludeText("Sure?");
  expect(wrapper.find(".btn").filterWhere(d => d.text() === "Sure!")).toExist();
});
