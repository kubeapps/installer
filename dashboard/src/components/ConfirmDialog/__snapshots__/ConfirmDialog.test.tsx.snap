// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading is true loading spinner matches the snapshot 1`] = `
<Fragment>
  <CdsModal
    closable={true}
    onCloseChange={[MockFunction]}
    size="default"
  >
    <div
      className="center"
    >
      <CdsModalContent>
        <p>
          Loading, please wait
        </p>
        <div
          className="center"
        >
          <LoadingWrapper
            loaded={false}
          />
        </div>
      </CdsModalContent>
    </div>
  </CdsModal>
</Fragment>
`;
